app/controllers/oligrapher_controller.rb

```diff
@@ -94,6 +94,27 @@ def find_nodes
     render json: entities
   end
 
+  def find_connections
+    return head :bad_request if params[:entity_id].blank?
+
+    @entity = Entity.unscoped.find(params[:entity_id])
+    raise Entity::EntityDeleted if @entity.is_deleted?
```

Comment:
If an entity doesn't exist, an ActiveRecord error is thrown and a 404 is automatically returned. Unless it's needed to raise this specific error here, you can probably remove this line and remove the `unscoped` call in the line above

Reply:
Makes sense. I just took the code directly from the `connections` and `set_entity` methods in https://github.com/public-accountability/littlesis-rails/blob/master/app/controllers/api/entities_controller.rb

Reply:
ah right. One thing to that's easy to overlool: The `Api::EntitiesController` subclasses  [ApiController](https://github.com/public-accountability/littlesis-rails/blob/master/app/controller/api/api_controller.rb) while `EntitiesController` subclasses [ApplicationController]( https://github.com/public-accountability/littlesis-rails/blob/master/app/controllers/application_controller.rb), which has the method [set_entity](https://github.com/public-accountability/littlesis-rails/blob/master/app/controllers/application_controller.rb#L110-L112). However it uses the parameter `id` instead of `entity_id` so it can't be used unless you adapt it to use the parameter `entity_id` as well.